sql/rpl_mi.cc

```diff
@@ -197,142 +180,20 @@ void Master_info::clear_in_memory_info(bool all)
   }
 }
 
-
-const char *
-Master_info::using_gtid_astext(enum enum_using_gtid arg)
-{
-  switch (arg)
-  {
-  case USE_GTID_NO:
-    return "No";
-  case USE_GTID_SLAVE_POS:
-    return "Slave_Pos";
-  default:
-    DBUG_ASSERT(arg == USE_GTID_CURRENT_POS);
-    return "Current_Pos";
-  }
-}
-
-
 void init_master_log_pos(Master_info* mi)
 {
   DBUG_ENTER("init_master_log_pos");
-
   mi->master_log_name[0] = 0;
   mi->master_log_pos = BIN_LOG_HEADER_SIZE;             // skip magic number
-  if (mi->master_supports_gtid)
-  {
-    mi->using_gtid= Master_info::USE_GTID_SLAVE_POS;
-  }
+  mi->master_use_gtid.set_default();
+  mi->master_heartbeat_period.set_default();
```

Comment:
There was no “USE_GTID_DEFAULT” previously, so `master_use_gtid` was reset only conditionally.
`master_heartbeat_period`, OTOH, was reset in CHANGE MASTER rather than RESET REPLICA.

I wonder, though, why doesn’t `RESET REPLICA ALL` reset all `CHANGE MASTER` fields of the default unnamed connection, whereas it is a complete deletion for named (as in multi-source) connections?